### PR TITLE
Add test for String tensor with middle null character

### PR DESCRIPTION
**Pull request by CoderHam**
- [x] Fix TF output string tensor calculation
- [x] Add test cases for graphdef and savedmodel
- [ ] Add Onnx after backend is fixed

### CODE REVIEW

File: qa/L0_nullchar_string/test.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+RET=0
+
+set +e
+
+python $NULLCHAR_CLIENT_PY -m graphdef_nobatch_zero_1_object -v >>$CLIENT_LOG 2>&1
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_nullchar_string/test.sh]: Sure, will fix this

File: src/backends/tensorflow/base_backend.cc [hunk 2]
```diff
@@ -633,6 +634,7 @@ BaseBackend::Context::SetStringInputTensor(
 
       TRTISTF_TensorSetString(
           tensor, tensor_element_idx + element_idx, content, len);
+      std::string tmp = std::string(content, len);
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply src/backends/tensorflow/base_backend.cc]: Used this for logging. Removed now.

File: qa/L0_nullchar_string/test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_nullchar_string/test.sh]: We don't currently have a custom string model to test string inputs to the custom backend (cannot use addsub with a nullcharacter).
I can create a ticket for this and add it to all the places where we should have it but as I understand the user implements the custom backend and is responsible for the string parsing himself. Hence I am unsure why we need to test this ourselves

File: qa/L0_nullchar_string/nullchar_string_client.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    input0_data = np.array([tmp_str], dtype=object)
+
+    # Send inference request to the inference server. Get results for
+    # both output tensors.
+    result = ctx.run({ 'INPUT0' : (input0_data,) },
+                     { 'OUTPUT0' : InferContext.ResultFormat.RAW },
+                     batch_size)
+
+    # We expect there to be 2 results (each with batch-size 1). Walk
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_nullchar_string/nullchar_string_client.py]: Fixed

File: qa/L0_nullchar_string/nullchar_string_client.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
```
Comment (GuanLuo) [inline qa/L0_nullchar_string/nullchar_string_client.py]: "#"

Comment (deadeyegoodwin) [reply qa/L0_nullchar_string/nullchar_string_client.py]: Just delete the blank line to match standard style

File: qa/L0_nullchar_string/test.sh [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+RET=0
+
+set +e
+
+# Add onnx_nobatch_zero_1_object once Onnx backend is fixed
```
[the remark this answers is not in the document]
Comment (CoderHam) [reply qa/L0_nullchar_string/test.sh]: Yes, spoke to David about this. Already filed a ticket for this.

File: qa/L0_nullchar_string/nullchar_string_client.py (hunk 4 above)
Comment (CoderHam) [reply qa/L0_nullchar_string/nullchar_string_client.py]: Done
